tests/test_rate_parsing.py

```
@@ -0,0 +1,142 @@
+        """'0%' must parse to 0.0."""
+        assert self.guard._parse_rate("0%") == 0.0
+
+    def test_whitespace_handling(self):
+        """'  5.25%  ' must parse correctly with whitespace."""
+        assert self.guard._parse_rate("  5.25%  ") == 0.0525
+
+    def test_invalid_input_raises(self):
+        """Non-numeric input must raise ValueError (fail-closed)."""
+        try:
+            self.guard._parse_rate("abc")
+            assert False, "Should have raised ValueError"
```

Ruff (0.15.12)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)